### DISCUSSION TITLE
Datetime picker cannot be set to null

### DISCUSSION THREAD

**Discussion by ZacharyDuBois** [filamentphp/filament#2311] [Help]
### Package

filament/forms

### Package Version

v2.10.44

### Laravel Version

v9.8.1

### Livewire Version

v2.10.5

### PHP Version

8.1.4

### Bug description

DateTime field does not persist null values (even with nullable set).

### Steps to reproduce

Migration:

```php
//

Schema::create('tests', function (Blueprint $table) {
    $table->uuid('id')
        ->primary();
    $table->timestamp('testing_at')
        ->nullable()
        ->index();
});

//
```

Model:

```php
class Test extends Model
{
    use HasUUIDPrimaryKey

    protected $fillable = ['testing_at'];

    protected $casts = [
        'testing_at' => 'datetime',
    ];
}
```

Filament resource:
```php
//

public static function form(Form $form): Form
{
    return $form
        ->schema([
            DateTimePicker::make('pick_available_at')
                ->nullable(),
        ]);
}

//
```

### Relevant log output

```shell
N/A
```

**Comment by ryangjchandler** [comment]
@ZacharyDuBois The column isn't nullable in the migration?

**Comment by ZacharyDuBois** [comment]
@ryangjchandler My apologies I copied and changed the field name of the wrong column. Updated my steps to reproduce.

**Comment by danharrin** [comment]
Cannot reproduce this, sorry 🤷‍♂️

**Comment by TiiFuchs** [comment]
I have the same problem with Filament v4.3.1

The column is nullable, (its initial value is null).
The DatePicker is nullable too.

But if you remove the day,month and year and save, and then reload the page it's back to the previous value. It does not get removed (set to null) in the database.

**Reply by TiiFuchs** [reply]
I think something somewhere is just incorrectly configured... But I'm just not able to find it. :/

**Reply by TiiFuchs** [reply]
Okay, the problem was that the datetime fields were not listed in the $fillable array on my model.
